crypto/bio/bss_conn.c

```diff
@@ -198,7 +198,7 @@ static int conn_state(BIO *b, BIO_CONNECT *c)
             ret = BIO_connect(b->num, BIO_ADDRINFO_address(c->addr_iter),
                               BIO_SOCK_KEEPALIVE | c->connect_mode);
             b->retry_reason = 0;
-            if (ret < 0) {
+            if (ret == 0) {
                 if (BIO_sock_should_retry(ret)) {
```

Comment:
Ok I modify it and I create a new pull req?

2016-04-18 16:47 GMT+02:00 Richard Levitte <notifications@github.com>:

> In crypto/bio/bss_conn.c
> <https://github.com/openssl/openssl/pull/972#discussion_r60071401>:
>
> > @@ -198,7 +198,7 @@ static int conn_state(BIO *b, BIO_CONNECT *c)
> >              ret = BIO_connect(b->num, BIO_ADDRINFO_address(c->addr_iter),
> >                                BIO_SOCK_KEEPALIVE | c->connect_mode);
> >              b->retry_reason = 0;
> > -            if (ret < 0) {
> > +            if (ret == 0) {
> >                  if (BIO_sock_should_retry(ret)) {
>
> The argument to BIO_sock_should_retry should be -1 rather than ret.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openssl/openssl/pull/972/files/b4e505270a1737f9afc221ded83603cf1cf84551#r60071401>
>